runtime/shiden/src/precompiles.rs

```diff
@@ -63,6 +65,31 @@ impl Contains<RuntimeCall> for WhitelistedCalls {
         }
     }
 }
+/// Filter that only allows whitelisted runtime call to pass through dispatch-lockdrop precompile
+pub struct WhitelistedLockdropCalls;
+
+impl Contains<RuntimeCall> for WhitelistedLockdropCalls {
+    fn contains(t: &RuntimeCall) -> bool {
+        match t {
+            RuntimeCall::Utility(pallet_utility::Call::batch { calls })
+            | RuntimeCall::Utility(pallet_utility::Call::batch_all { calls }) => calls
+                .iter()
+                .all(|call| WhitelistedLockdropCalls::contains(call)),
```

Comment:
This looks like unbounded recursion.

Reply:
I've added one unit test to ensure the iteration in batch calls is actually bounded, and I've changed the max bytes len of a call to a more conservative value (divided by 2: 2048).
Thanks @Dinonard